src/uu/ls/src/ls.rs

```diff
@@ -2238,6 +2238,7 @@ fn display_items(items: &[PathData], config: &Config, out: &mut BufWriter<Stdout
             for item in items {
                 let context_len = item.security_context.len();
                 longest_context_len = context_len.max(longest_context_len);
+                item.md(out);
```

Comment:
I think a comment explaining why this is here would be nice. I'm also thinking maybe it makes more sense to fix this in the `get_security_context` function?

Reply:
I've looked at inserting something similar into `get_security_context()` but I really wanted to use the `md()` function from PathData as this is the "code path" that is taken in the case of `ls -L -l` and which generates the error.

At a first glance the patch looked a bit more complex to implement it inside `get_security_context()` as I don't have direct access to a `PathData()` or the `out` variable from `get_security_context()`.

I can try to make it happen with a bit of rework though :)

Reply:
Just attempting to dereference in the `get_security_context` function should work right?

Reply:
Essentially what I'm proposing is this:
- If security context is supported and the feature enabled: get the context and return the result
- If not and `must_dereference` is true:
    - perform a deref, if it fails, return that error
    - return Ok("?".into())
- Else just return Ok("?".into())